pkg/scheduler/core/generic_scheduler.go

```diff
@@ -93,7 +93,8 @@ func (g *genericScheduler) findClustersThatFit(
 	defer metrics.ScheduleStep(metrics.ScheduleStepFilter, time.Now())
 
 	var out []*clusterv1alpha1.Cluster
-	clusters := clusterInfo.GetReadyClusters()
```

Comment:
According to https://github.com/karmada-io/karmada/issues/1853#issuecomment-1141593401,

You mentioned that your cluster  `is default cordoned`, does that means you will set the taint(key: `cluster.karmada.io/unschedulable`, effect=`NoSchedule`) to the cluster? Just similar thing like `karmadactl taint`  do. 

My question here is even if a cluster has taints, it won't be filtered by `GetReadyClusters()` as the taint won't lead to condition change, right? 

I'm still confused about the real use case.

Reply:
User story:

There are k8s resources which need to be dispatched to member clusters filtered by selector before member cluster can serve. Just like daemonset for single cluster.

There are two limit in karmada

1. `karmada-scheduler` will not trigger a schedule cycle if the object has been scheduled successfully once even if there are new clusters which satisfied the pp's labelSelector added. 
2. `karmada-scheduler` will not trigger a schedule cycle if the object has been scheduled successfully once even if  there are some clusters whose status changed from notready to ready. 

So the object can't be dispatched to clusters new added or status changed from notready to ready. 

For case 1: we added a custom controller to update rb directly.
For case 2: we use this pr to let the notready clusters can also be set to `rb.spec.clusters` and work will be created. When the cluster's status changes from notready to ready, the work can be executed successfully.